Add jenkins_job_launcher role (#257)
redhatci/ansible-collection-redhatci-ocp: roles/jenkins_job_launcher/README.md

# jenkins_job_launcher

jenkins_job_launcher role triggers Jenkins job using the uri module.

> the role makes use of username + user API token to authenticate with Jenkins.

Reviewer: ```suggestion
This role uses username + user API token to authenticate with Jenkins.
```

Role Variables
--------------

> Required variables are: 

Reviewer: Could this be in a nice table like [this](https://github.com/redhatci/ansible-collection-redhatci-ocp/tree/main/roles/create_pr)?

job_url - Jenkins job url
job_params - job params and values based on the following playbook example.
username - Jenkins user

> token - the user API token

Reviewer 2: We don't enforce [role variable naming through linting check](https://github.com/redhatci/ansible-collection-redhatci-ocp/blob/main/.ansible-lint#L15) but we do require all new roles to follow some sort of role variable naming. For example, role variable linting expects variables with the prefix `jenkins_job_launcher_` but we understand it could be quite long, in those cases using something like `jjl_` is good enough.

So `job_url` becomes `jjl_job_url` or perhaps `jjl_url`, etc.

          
Example Playbook
----------------

>   hosts: all

Reviewer: I think this should be in a code block for better readability

  tasks: 
  - name: trigger jenkins job
    ansible.builtin.import_role: 

>       jenkins_job_launcher 

Reviewer 2: ```suggestion
      redhatci.ocp.jenkins_job_launcher 
```

# jenkins_job_launcher

> ```

Reviewer: I think the code block should be just for the example playbook
